pkg/apis/core/validation/validation.go

```diff
@@ -1532,12 +1532,22 @@ func validateStorageOSPersistentVolumeSource(storageos *core.StorageOSPersistent
 
 // validatePVSecretReference check whether provided SecretReference object is valid in terms of secret name and namespace.
 
-func validatePVSecretReference(secretRef *core.SecretReference, fldPath *field.Path) field.ErrorList {
+func validatePVSecretReference(secretRef *core.SecretReference, allowDNSSubDomainFormat bool, fldPath *field.Path) field.ErrorList {
 	var allErrs field.ErrorList
 	if len(secretRef.Name) == 0 {
 		allErrs = append(allErrs, field.Required(fldPath.Child("name"), ""))
 	} else {
-		allErrs = append(allErrs, ValidateDNS1123Label(secretRef.Name, fldPath.Child("name"))...)
+		if !allowDNSSubDomainFormat {
```

Comment:
Nit: don't negate `allowDNSSubDomainFormat`, it's hard to read. Switch `then` / `else` branches below.

Reply:
sure, done